packages/opencensus-exporter-stackdriver/src/stackdriver-stats-utils.ts

```diff
@@ -0,0 +1,288 @@
+      valueType: ValueType): Point {
+    let value;
+    if (valueType === ValueType.INT64) {
+      value = {int64Value: point.value as number};
+    } else if (valueType === ValueType.DOUBLE) {
+      value = {doubleValue: point.value as number};
+    } else if (valueType === ValueType.DISTRIBUTION) {
+      value = {
+        distributionValue:
+            this.createDistribution(point.value as DistributionValue)
+      };
+    } else {
+      // console.log(`${valueType} is not supported.`);
+    }
+
+    const endTime = this.toISOString(point.timestamp);
+    if (startTimeStamp) {
+      // Must be present for cumulative metrics.
+      const startTime = this.toISOString(startTimeStamp);
+      return {interval: {startTime, endTime}, value};
+    }
+
+    return {interval: {endTime}, value};
+  }
+
+  /**
+   * Formats an OpenCensus Distribution to Stackdriver's format.
+   * @param distribution The OpenCensus Distribution Data.
+   */
+  static createDistribution(distribution: DistributionValue): Distribution {
+    return {
+      count: distribution.count,
+      mean: distribution.count === 0 ? 0 :
+                                       distribution.sum / distribution.count,
+      sumOfSquaredDeviation: distribution.sumOfSquaredDeviation,
+      bucketOptions: {
+        explicitBuckets: {
+          bounds: this.createExplicitBucketOptions(distribution.bucketOptions)
+        }
+      },
+      bucketCounts: this.createBucketCounts(distribution.buckets)
+    };
+  }
+
+  /**
+   * Converts a OpenCensus BucketOptions to a StackDriver BucketOptions.
+   * @param buckets The DistributionValue BucketOptions.
+   */
+  static createExplicitBucketOptions(bucketOptions: BucketOptions): number[] {
+    const explicitBucketOptions: number[] = [];
+    // The first bucket bound should be 0.0 because the Metrics first bucket is
+    // [0, first_bound) but Stackdriver monitoring bucket bounds begin with
+    // -infinity (first bucket is (-infinity, 0))
+    explicitBucketOptions.push(0);
+    return explicitBucketOptions.concat(bucketOptions.explicit.bounds);
+  }
+
+  /**
+   * Converts a OpenCensus Buckets to a list of counts.
+   * @param buckets The DistributionValue buckets.
+   */
+  static createBucketCounts(buckets: DistributionBucket[]): number[] {
+    const bucketCounts: number[] = [];
+    // The first bucket (underflow bucket) should always be 0 count because the
+    // Metrics first bucket is [0, first_bound) but StackDriver distribution
+    // consists of an underflow bucket (number 0).
+    bucketCounts.push(0);
+    buckets.map((bucket: DistributionBucket) => {
+      bucketCounts.push(bucket.count);
+    });
+    return bucketCounts;
+  }
+
+  private static toISOString(timestamp: Timestamp) {
+    const str = new Date(timestamp.seconds * 1000).toISOString();
+    const nsStr = `${this.padNS(timestamp.nanos)}`.replace(/0+$/, '');
+    return str.replace('000Z', `${nsStr}Z`);
+  }
+
+  private static padNS(ns: number) {
```

Comment:
Could you make this function name a bit more descriptive? It seems like it's left padding a number, but I wouldn't know that without reading the code.